dnscache/manager.go

```diff
@@ -96,6 +105,21 @@ func (m *DnsCacheManager) doCachedDial(d *net.Dialer, ctx context.Context, netwo
 		return safeDial(address, "")
 	}
 
+	if m.strategy == config.NoCacheStrategy {
+		if len(ips) > 1 {
+			m.cacheStorage.Delete(host)
+			return safeDial(ips[0]+":"+port, "")
```

Comment:
what is the difference between `random` and `pick_first` then?

Reply:
You probably meant what is the difference between "pick first" and "no_cache". 
In think idea here, that since `m.cacheStorage.Delete(host)` is run, it forces "cache" wrapper to run DNS every time. Additionally "pick first" is a default Go DNS strategy too. In other words, we first cache and then disregard cache immediately.

Reply:
ah right, I meant difference between "pick_first" and "no_cache".

I see here that our cache logic is still influences the choice of which IP to pick. Go DNS strategy depends on env var `GODEBUG=netdns=`. There probably won't be any Go resolver in place.

Reply:
True, but the alternative, in this case, is do a second DNS call, and run `safeDial(host, "")` 😛

Reply:
I think I need to clarify:

I mean if we leave multiple IPs entries in cache - it will do real resolve only once to put it in cache then if item from cache is multi-IPs we do `return safeDial(address, "")` - that would guarantee that `go` or `cgo` (whatever set to be used) is making choice which IP to pick.

also, in this case we caching, but caching (with TTL) only that fact that given host has multiple IPs

Reply:
@buger tbh I am OK with this implementation

just wanted to mention that it is no_cache but still `pick_first` at the same time, discarding system dealer logic for multiple IPs. if we are Ok with this let's leave it.

Reply:
@dencoded I think you missed some point:
>that would guarantee that `go` or `cgo` (whatever set to be used) is making choice which
It won't, because it will have a list with 1 ip and this is the only ip which will be used for connection establishment(in other words - no additional dns request would take place). See [sources](https://golang.org/src/net/lookup.go#L198): LookupIPAddr is called before [connecting](https://golang.org/src/net/dial.go#L402)

@buger FYI

Reply:
@evityuk my point was that for multiple IPs we need to call `net.Dial` passing `address` (not IP) and it will do `LookupIPAddr` inside applying its system/default logic
instead of:
doing call `net.Dial` with 1st IP picked by us and eliminating any logic inside standard `net.Dial`